lib/offline-handling.ts

```
@@ -1,23 +1,55 @@
-// Function to handle offline requests with retry logic
+/**
+ * Enhanced function to handle offline requests with retry logic and better error tracking
+ */
 export async function withOfflineHandling<T>(
   callback: () => Promise<T>,
   options: {
     retries?: number;
     retryDelay?: number;
     fallback?: T;
     onOffline?: () => void;
+    onRetry?: (attempt: number, error: any) => void;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)